src/manifest.c

```diff
@@ -683,34 +689,48 @@ struct manifest *load_manifest(int current, int version, struct file *file, stru
 {
 	struct manifest *manifest = NULL;
 	int ret = 0;
+	int delta_manifest_version = 0;
+	bool delta_manifest = mom->use_delta_manifest;
 	bool retried = false;
 
 retry_load:
-	ret = retrieve_manifests(current, version, file->filename, file, file->is_mix);
-	if (ret != 0) {
-		fprintf(stderr, "Failed to retrieve %d %s manifest\n", version, file->filename);
-		return NULL;
-	}
+	if (delta_manifest && file->peer != NULL) {
+		delta_manifest_version = file->peer->last_change;
+	} else {
+		delta_manifest = false;
+		delta_manifest_version = 0;
 
-	if (!header_only) {
-		ret = verify_bundle_hash(mom, file);
-	}
+		ret = retrieve_manifests(current, version, file->filename, file, file->is_mix);
+		if (ret != 0) {
+			fprintf(stderr, "Failed to retrieve %d %s manifest\n", version, file->filename);
+			return NULL;
+		}
 
-	if (ret != 0) {
-		if (retried == false) {
-			remove_manifest_files(file->filename, version, file->hash);
-			retried = true;
-			goto retry_load;
+		if (!header_only) {
+			ret = verify_bundle_hash(mom, file);
+		}
+
+		if (ret != 0) {
+			if (retried == false) {
+				remove_manifest_files(file->filename, version, file->hash);
+				retried = true;
+				goto retry_load;
+			}
+			return NULL;
 		}
-		return NULL;
 	}
 
-	manifest = manifest_from_file(version, file->filename, header_only, false, file->is_mix);
+	manifest = manifest_from_file(version, delta_manifest_version, file->filename, header_only, false, file->is_mix);
 
 	if (manifest == NULL) {
 		if (retried == false) {
-			remove_manifest_files(file->filename, version, file->hash);
-			retried = true;
+			if (delta_manifest) {
+				delta_manifest = false;
+				delta_manifest_version = 0;
+			} else {
+				remove_manifest_files(file->filename, version, file->hash);
+				retried = true;
+			}
```

Comment:
ok, so if manifest failed to load and retried is false, we set delta_manifest to false, delta_manifest_version to 0 (essentially turning off delta_manifest options for this manifest) and retry without setting retried to true. This means we will try to download the full manifest and still allow it to retry one more time.

This flow is good, but please add a lengthy comment explaining the flow. It took me a while to figure that out just by reading the code here.